packages/integration_test/test/integration_test_driver_extended_test.dart

```diff
@@ -0,0 +1,117 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_driver/flutter_driver.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test_driver_extended.dart';
+import 'package:webdriver/sync_io.dart';
+
+void main() {
+  group('integrationDriver', () {
+    late FakeFlutterWebConnection fakeConnection;
+    late WebFlutterDriver driver;
+    late bool called;
+    late int exitCode;
+
+    setUpAll(() {
+      fakeConnection = FakeFlutterWebConnection();
+      driver = WebFlutterDriver.connectedTo(fakeConnection);
+      exitFn = (int code) => exitCode = code;
+    });
+
+    setUp(() {
+      called = false;
+      exitCode = -1;
+    });
+
+    test('write response data when all test pass', () async {
+      fakeConnection.fakeResponse = r'''
+{
+  "isError": false,
+  "response": {
+    "message": "{\"result\": \"true\", \"data\": {\"reports\": \"passed\"}}"
+  }
+}
+''';
+
+      await integrationDriver(
+        driver: driver,
+        responseDataCallback: (_) {
+          called = true;
+        },
+      );
+      expect(called, true);
+      expect(exitCode, 0);
+    });
+    test(
```

Comment:
nit (here and below): separate test cases by a blank line for easier readability.

Reply:
added